cmake/CeleritasUtils.cmake

```
@@ -64,6 +64,39 @@ CMake configuration utility functions for Celeritas.
   The ``<input>`` must be a relative path to the current source directory, and
   the ``<output>` path is configured to the project build "include" directory.
 
+.. command:: celeritas_setup_option
+
+  Add a single compile time option value to the list::
+
+    celeritas_setup_option(<var> <option> [conditional])
+
+  This appends ``<option>`` to the list ``<var>_OPTIONS`` if ``${conditional}``
+  is true or the argument is not provided; and to ``<var>_DISABLED_OPTIONS`` if
+  the variable is present and false.
+
+.. command:: celeritas_define_options
+
+  Set up CMake variables for defining configure-time option switches::
+
+    celeritas_define_options(<var> <doc>)
+
+  This will set <var> to the first item of the list ``${<var>_OPTIONS}``. It
+  will validate that the selection is one of the list and default to the first
+  item in the list.
+
+.. command:: celeritas_generate_option_macros
+
+  Generate preprocessor macros for the given option list.::
+
+    celeritas_generate_option_macros(<var>)
```

```suggestion
    celeritas_generate_option_definitions(<var>)
```
CMake also has the concept of [macros](https://cmake.org/cmake/help/latest/command/macro.html) and thus this was my first thought (until I read the documentation with indeed talks about `C preprocessor` in passing.  

The propose change is keeping in tone with CMake itself, for example [add_definition](https://cmake.org/cmake/help/latest/command/add_definitions.html)